### ISSUE TITLE
cleanup code

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/numpyro#276]
Just make a list here for some changes which have been put upstream to JAX to simplify our code base.
- np.var has `ddof` support
- np.median is implemented and has batching rule
- np.cov is available
- _elemwise_no_params functionality is available in jax
- various nonlinearity at stax is available: stax.Sigmoid (so we can revise vae example with this), slax.Elu
- entr/multigammaln is available at jax.scipy.special
- np.tril_indices is available
- lax.map can be used in sequential mcmc
